src/Facets/GenericSwapFacet.sol

```diff
@@ -3,42 +3,328 @@ pragma solidity 0.8.17;
+    /// @param _referrer the address of the referrer
+    /// @param _receiver the address to receive the swapped tokens into (also excess tokens)
+    /// @param _minAmountOut the minimum amount of the final asset to receive
+    /// @param _swapData an object containing swap related data to perform swaps before bridging
+    function swapTokensSingleERC20ToNative(
+        bytes32 _transactionId,
+        string calldata _integrator,
+        string calldata _referrer,
+        address payable _receiver,
+        uint256 _minAmountOut,
+        LibSwap.SwapData calldata _swapData
+    ) external payable {
+        // deposit funds
+        IERC20(_swapData.sendingAssetId).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _swapData.fromAmount
+        );
+
+        // ensure that contract (callTo) and function selector are whitelisted
+        if (
+            !(LibAllowList.contractIsAllowed(_swapData.callTo) &&
+                LibAllowList.selectorIsAllowed(bytes4(_swapData.callData[:4])))
+        ) revert ContractCallNotAllowed();
+
+        // get initial token balance
+        uint256 initialBalance = address(this).balance;
+
+        // check if the current allowance is sufficient
+        if (
+            IERC20(_swapData.sendingAssetId).allowance(
+                address(this),
+                _swapData.approveTo
+            ) < _swapData.fromAmount
+        ) {
+            // allowance insufficient - register max approval
+            SafeERC20.safeApprove(
+                IERC20(_swapData.sendingAssetId),
+                _swapData.approveTo,
+                0
+            );
+            SafeERC20.safeApprove(
+                IERC20(_swapData.sendingAssetId),
+                _swapData.approveTo,
+                type(uint256).max
+            );
+        }
+
+        // execute swap
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory res) = _swapData.callTo.call(
+            _swapData.callData
+        );
+        if (!success) {
+            string memory reason = LibUtil.getRevertMsg(res);
+            revert(reason);
+        }
+
+        // get final balance
+        uint256 amountReceived = address(this).balance - initialBalance;
+
+        // ensure that minAmountOut was received
+        if (amountReceived < _minAmountOut)
+            revert CumulativeSlippageTooHigh(_minAmountOut, amountReceived);
+
+        // transfer funds to receiver
+        // solhint-disable-next-line avoid-low-level-calls
+        (success, ) = _receiver.call{ value: amountReceived }("");
+        if (!success) revert NativeAssetTransferFailed();
+
+        // emit events (both required for tracking)
+        emit AssetSwapped(
+            _transactionId,
+            _swapData.callTo,
+            _swapData.sendingAssetId,
+            _swapData.receivingAssetId,
+            _swapData.fromAmount,
+            _minAmountOut,
```

Comment:
In `LibSwap` we emit here:
```
newBalance > initialReceivingAssetBalance
  ? newBalance - initialReceivingAssetBalance
  : newBalance
```

To have the same expected value it should be `amountReceived` here.

Reply:
Good spotting...updated.